#! /bin/sh /usr/share/dpatch/dpatch-run
## 74_admin_non-ascii_emails.dpatch by  <lionel@mamane.lu>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't assume emails are ASCII

@DPATCH@
diff -urNad mailman-2.1.5~/Mailman/Cgi/admin.py mailman-2.1.5/Mailman/Cgi/admin.py
--- mailman-2.1.5~/Mailman/Cgi/admin.py	2003-12-24 18:27:45.000000000 +0100
+++ mailman-2.1.5/Mailman/Cgi/admin.py	2005-11-14 00:51:45.409005398 +0100
@@ -867,7 +867,7 @@
     chunksz = mlist.admin_member_chunksize
     # The email addresses had /better/ be ASCII, but might be encoded in the
     # database as Unicodes.
-    all = [_m.encode() for _m in mlist.getMembers()]
+    all = mlist.getMembers()
     all.sort(lambda x, y: cmp(x.lower(), y.lower()))
     # See if the query has a regular expression
     regexp = cgidata.getvalue('findmember', '').strip()
